onadata/libs/tests/serializers/test_user_profile_serializer.py

```diff
@@ -103,3 +103,33 @@ def test_metadata_view_for_owner_only(self):
                          bob_serializer.data['username'])
         self.assertEqual(alice_profile.user.username,
                          alice_serializer.data['username'])
+
+    def test_username_not_allowed_if_numeric(self):
+        """
+        Test ValidationError raised for numeric usernames
+        """
+        # create a form instance with a purely numeric username
+        request = APIRequestFactory().get('/')
+        user_data = {
+            'username': '1234',
+            'email': 'numerica@example.ke',
+            'password1': 'num123',
+            'password2': 'num123',
+            'name': 'Num',
+            'city': 'Nai',
+            'country': 'KE',
+            'organization': '',
+            'home_page': '',
+            'twitter': ''
+        }
+        user_prof = self._create_user_profile(extra_post_data=user_data)
+        request.user = user_prof.user
```

Comment:
Doesn't this create a user with the username `1234` either way ?  Seems we want to block it at the serializer level; which is cool but I think it's weird to have a test do the exact thing it wants to block in a roundabout way...